Core/MIPS/MIPSAnalyst.cpp

```diff
@@ -1066,14 +1041,32 @@ namespace MIPSAnalyst {
 			if (end) {
 				currentFunction.end = addr + 4;
 				currentFunction.isStraightLeaf = isStraightLeaf;
+
+				// Check if we already have symbol info starting here.  If so, skip insertion.
+				// We used to use the symbols to find the functions, but sometimes we'd find
+				// wrong ones due to two modules with the same name.
+				u32 existingSize = g_symbolMap->GetFunctionSize(currentFunction.start);
+				if (existingSize != SymbolMap::INVALID_ADDRESS) {
+					currentFunction.foundInSymbolMap = true;
+
+					// If we run into a func with a different size, skip updating the hash map.
+					// This will prevent us saving incorrectly named funcs with wrong hashes.
```

Comment:
Yes, but also if *any* of the sizes are different.  So if there are 100 functions and only 2 of them differ in size, it will be enough to prevent it from saving the symbol map names with the new hashes.

It's not completely perfect though, yeah.

-[Unknown]